### PR TITLE
Add dark theme styling to Project About pages

### PR DESCRIPTION
**Pull request by goplayoutside3**
_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

Package:
app-project

Describe your changes:
I added the `ThemeModeToggle` to all Project About pages along with appropriate styling for dark theme.

There are some linter corrections included for double-quotes vs single-quotes.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by goplayoutside3** [comment]
@srallen and @mcbouslog I think you mentioned accessibility concerns when using `neutral-1` text over `accent-1` background. I noticed that `AboutNavLink` was already using those two colors together based on the light theme wireframes. Do you think we should change the colors a bit for more contrast?

File: packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const AboutDropContent = ({ aboutNavLinks, router }) => {
   const baseUrl = `/projects/${owner}/${project}/about`
   
   return (
-    <Nav gap="xsmall" background="white">
+    <Nav gap='xsmall' background={{ dark: 'dark-3', light: 'neutral-6' }}>
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.js:17]: The `dark-3` color background here is a bit hard to distinguish from the rest of the page. In the past, Becky has had us use `dark-5` in these cases and you can see that in action with the shared components' `Modal`. 

Here's how this looks:

Screen Shot 2021-07-19 at 11 00 37 AM

```suggestion
    <Nav gap='xsmall' background={{ dark: 'dark-5', light: 'neutral-6' }}>
```

File: packages/app-project/src/screens/ProjectAboutPage/components/AboutNavLink.js [hunk 2]
```diff
@@ -23,12 +23,12 @@ const AboutNavLink = ({ router, link }) => {
 
   return (
     <Box
-      background={isCurrentPage ? 'accent-1' : 'neutral-6'}
+      background={isCurrentPage ? 'accent-1' : { light: 'neutral-6' }}
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/components/AboutNavLink.js:26]: Similar comment here about the color object.

```suggestion
      background={isCurrentPage ? 'accent-1' : { light: 'neutral-6', dark: '' }}
```

File: packages/app-project/src/screens/ProjectAboutPage/components/TeamMember.js [hunk 3]
```diff
@@ -66,27 +68,27 @@ const TeamMember = ({ user, router }) => {
...
           />
         ) : (
-          <Image alt={user.display_name} fit="cover" src={user.avatar_src} />
+          <Image alt={user.display_name} fit='cover' src={user.avatar_src} />
         )}
       </StyledAvatar>
-      <Box flex direction="column">
-        <StyledDisplayName>{user.display_name}</StyledDisplayName>
+      <Box flex direction='column'>
+        <StyledDisplayName color={{ light: 'neutral-7' }}>{user.display_name}</StyledDisplayName>
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/components/TeamMember.js:84]: Similar comment here:

```suggestion
        <StyledDisplayName color={{ light: 'neutral-7', dark: '' }}>{user.display_name}</StyledDisplayName>
```

**Review by srallen — CHANGES_REQUESTED** [review]
I have a few minor suggestions. One thing to look out for is that when you use an object to set the colors, Grommet will expect something set for both the light and dark properties, otherwise some invalid CSS could be generated for the browser. In the cases where nothing should be set for the dark theme, then an empty string can be set. 

Otherwise, this is looking really good. Thanks!

File: packages/app-project/src/screens/ProjectAboutPage/components/AboutNavLink.js [hunk 4]
```diff
@@ -23,12 +23,12 @@ const AboutNavLink = ({ router, link }) => {
...
   return (
     <Box
-      background={isCurrentPage ? 'accent-1' : 'neutral-6'}
+      background={isCurrentPage ? 'accent-1' : { light: 'neutral-6', dark: '' }}
       pad={{ horizontal: '20px', vertical: '5px' }}
     >
       <NavLink
         link={link}
-        color={isCurrentPage ? 'neutral-1' : 'dark-5'}
+        color={isCurrentPage ? 'neutral-1' : { dark: 'neutral-6', light: 'dark-5' }}
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/components/AboutNavLink.js:31]: I just realized I missed your question about this from my last review. I ran a lighthouse audit on this and interestingly it did not flag this as being an issue. However, storybook's accessibility auditor definitely did when these two colors were used together like this in the survey task. So I think to be safe, maybe we should just stay consistent with the original two set colors and not change this based on being on the current page?

```suggestion
        color={{ dark: 'neutral-6', light: 'dark-5' }}
```

As an aside, I realized also that we don't have stories for theses components, so this is not a super high priority, but whenever you have extra time, I think it'd be helpful to add a few key stories for this part of the app, maybe one for the side bar, one for the team list, and one for the main markdown content? We can just open an issue to track this for now.

**Review by srallen — APPROVED** [review]
I missed your question about the colors on the sidebar navlink, so I've left a suggestion for that, but otherwise this looks good! Thanks!

File: packages/app-project/src/screens/ProjectAboutPage/components/AboutNavLink.js (hunk 4 above)
Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectAboutPage/components/AboutNavLink.js:31]: Sounds good! I've opened #2317 to add stories for those components.

**Pull request merged**